verilog/analysis/flow_tree.h

```diff
@@ -0,0 +1,133 @@
+// Copyright 2017-2022 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_VERILOG_FLOW_TREE_H_
+#define VERIBLE_VERILOG_FLOW_TREE_H_
+
+#include <bitset>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "absl/status/status.h"
+#include "common/lexer/token_stream_adapter.h"
+#include "verilog/parser/verilog_token_enum.h"
+
+namespace verilog {
+
+// FlowTree class builds the control flow graph of a tokenized System-Verilog
+// source code. Furthermore, enabling doing the following queries on the graph:
+// 1) Generating all the possible variants (provided via a callback function).
+// 2) TODO(karimtera): uniquely identify a variant with a bitset.
+class FlowTree {
+ public:
+  using BitSet = std::bitset<128>;
+  using TokenSequenceConstIterator = verible::TokenSequence::const_iterator;
+
+  struct ConditionalBlock {
+    TokenSequenceConstIterator ifdef_iterator;
+    TokenSequenceConstIterator ifndef_iterator;
+    std::vector<TokenSequenceConstIterator> elsif_iterators;
+    TokenSequenceConstIterator else_iterator;
+    TokenSequenceConstIterator endif_iterator;
+  };
+
+  struct Variant {
+    // Contains the token sequence of the variant.
+    verible::TokenSequence sequence;
+
+    // The i-th bit in "macros_mask" is 1 when the macro (with ID = i) is
+    // assumed to be defined, o.w. it is assumed to be undefined.
```

Comment:
o.w. stands for otherwise ? This is an unusual abbreviation; either use `o/w` or write it out `otherwise`.